hessian dependency conflict

### Your question
seata 2.0使用了sofa-jraft做raft的集群部署,sofa-jraft使用了bolt做通信,当使用间接使用bolt进行rpc集群间通信时,由于seata所依赖的hessian版本为4.0.63,而sofa-bolt使用的是4.0.3
由于依赖冲突导致了无法正常使用相关功能,并且seata社区后续将会移除不再维护且漏洞较多的hessian,请问bolt社区这方面是否有相关替代产品计划,比如改用kryo,或者fury?
目前seata侧通过实现了个自定义的序列化器来处理该问题,但由于hessian是默认序列化器,在staic方法块中被加载,导致一定会触发异常,请问目前有什么好的办法能解决该问题吗?

```
Caused by: java.lang.InstantiationException: com.caucho.hessian.io.StackTraceElementDeserializer
	at java.lang.Class.newInstance(Class.java:427)
	at com.caucho.hessian.io.ContextSerializerFactory.init(ContextSerializerFactory.java:322)
	... 22 common frames omitted
Caused by: java.lang.NoSuchMethodException: com.caucho.hessian.io.StackTraceElementDeserializer.<init>()
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.newInstance(Class.java:412)
	... 23 common frames omitted
<==
```

### Your scenes

describe your use scenes (why need this feature)

### Your advice

describe the advice or solution you'd like

### Environment

- SOFABolt version:
- JVM version (e.g. `java -version`):
- OS version (e.g. `uname -a`):
- Maven version:
- IDE version:

--- a/src/main/java/com/alipay/remoting/serialization/SerializerManager.java
+++ b/src/main/java/com/alipay/remoting/serialization/SerializerManager.java
@@ -37,17 +37,20 @@ public class SerializerManager {
     private static final ReentrantLock REENTRANT_LOCK = new ReentrantLock();
 
     public static Serializer getSerializer(int idx) {
-        if (serializers[idx] == null && idx == Hessian2) {
+        Serializer currentSerializer = serializers[idx];
+        if (currentSerializer == null && idx == Hessian2) {
             REENTRANT_LOCK.lock();
             try {
-                if (serializers[idx] == null) {
-                    addSerializer(Hessian2, new HessianSerializer());
+                currentSerializer = serializers[idx];
+                if (currentSerializer == null) {
+                    currentSerializer = new HessianSerializer();
+                    addSerializer(Hessian2, currentSerializer);
                 }
             } finally {
                 REENTRANT_LOCK.unlock();
             }
         }
-        return serializers[idx];
+        return currentSerializer;
     }
 
     public static void addSerializer(int idx, Serializer serializer) {

--- a/src/main/java/com/alipay/remoting/serialization/SerializerManager.java
+++ b/src/main/java/com/alipay/remoting/serialization/SerializerManager.java
@@ -30,12 +30,12 @@ import java.util.concurrent.locks.ReentrantLock;
  */
 public class SerializerManager {
 
-    private static Serializer[] serializers = new Serializer[5];
-    public static final byte    Hessian2    = 1;
+    private static Serializer[]        serializers    = new Serializer[5];
+    public static final byte           Hessian2       = 1;
     //public static final byte    Json        = 2;
 
     private static final ReentrantLock REENTRANT_LOCK = new ReentrantLock();
-    
+
     public static Serializer getSerializer(int idx) {
         if (serializers[idx] == null && idx == Hessian2) {
             REENTRANT_LOCK.lock();

--- a/src/main/java/com/alipay/remoting/serialization/SerializerManager.java
+++ b/src/main/java/com/alipay/remoting/serialization/SerializerManager.java
@@ -16,6 +16,8 @@
  */
 package com.alipay.remoting.serialization;
 
+import java.util.concurrent.locks.ReentrantLock;
+
 /**
  * Manage all serializers.
  *
@@ -32,11 +34,19 @@ public class SerializerManager {
     public static final byte    Hessian2    = 1;
     //public static final byte    Json        = 2;
 
-    static {
-        addSerializer(Hessian2, new HessianSerializer());
-    }
-
+    private static final ReentrantLock REENTRANT_LOCK = new ReentrantLock();
+    
     public static Serializer getSerializer(int idx) {
+        if (serializers[idx] == null && idx == Hessian2) {
+            REENTRANT_LOCK.lock();
+            try {
+                if (serializers[idx] == null) {
+                    addSerializer(Hessian2, new HessianSerializer());
+                }
+            } finally {
+                REENTRANT_LOCK.unlock();
+            }
+        }
         return serializers[idx];
     }
 

